framework/src/multiapps/TransientMultiApp.C

```diff
@@ -210,16 +210,13 @@ TransientMultiApp::solveStep(Real dt, Real target_time, bool auto_advance)
         bool local_first = _first;
 
         // Now do all of the solves we need
-        while (true)
+        while (!at_steady && ex->getTime() + app_time_offset + 2e-14 < target_time)
```

Comment:
Yes agreed, This exact same condition with the same hardcoded tolerance appears elsewhere in this file too. What I'd really like to do is make additional callbacks so that we can execute the same logic as the executioner. Otherwise, this file will get out of sync again as we make more changes to `Transient.C`